server/src/main/java/org/apache/iotdb/db/mpp/memory/MemoryPool.java

```diff
@@ -79,8 +142,37 @@ public synchronized void free(String queryId, long bytes) {
     } else {
       queryMemoryReservations.put(queryId, queryReservedBytes);
     }
-
     reservedBytes -= bytes;
+
+    List<MemoryReservationFuture<Void>> futures =
```

Comment:
In this way, if one query free its memory, it will only notify blocking Futures belonging to this query. Other queries won't be notified and still waiting. 
And it seems that one free operation will only notify one blocking Future, but what if this freeing operation frees a large memory block which is large enough for multiple blocking Futures.

Reply:
Make sense. Will fix it.